content/recipes/terminus.md

```diff
@@ -163,125 +186,60 @@ The next thing we need to do is registering the health indicator as a provider.
 @@filename(app.module)
 import { Module } from '@nestjs/common';
 import { TerminusModule } from '@nestjs/terminus';
-import { TerminusOptionsService } from './terminus-options.service';
 import { DogHealthIndicator } from './dog.health';
 
 @Module({
-  imports: [
-    TerminusModule.forRootAsync({
-      imports: [ApplicationModule],
-      useClass: TerminusOptionsService,
-    }),
-  ],
-  providers: [DogHealthIndicator],
-  exports: [DogHealthIndicator],
+  controllers: [HealthController],
+  imports: [TerminusModule],
+  providers: [DogHealthIndicator]
 })
-export class ApplicationModule { }
+export class AppModule { }
 ```
 
-> info **Hint** In a real world application the `DogHealthIndicator` should be provided in a separate module, for example `DogsModule`, which then will be imported by the `ApplicationModule`. But keep in mind to add the `DogHealthIndicator` to the `exports` array of the `DogModule` and add the `DogModule` in `imports` array of the `TerminusModule.forRootAsync()` parameter object.
+> info **Hint** In a real-world application the `DogHealthIndicator` should be provided in a separate module, for example, `DogModule`, which then will be imported by the `AppModule`.
 
-The last required thing to do is to add the now available health indicator in the required health check endpoint. For that we go back to our `TerminusOptionsService` and implement it to the `/health` endpoint.
+The last required step we need to do is to add the now available health indicator in the required health check endpoint. For that, we go back to our `HealthController` and add it to our `check` function.
 
 ```typescript
-@@filename(terminus-options.service)
-import {
-  TerminusEndpoint,
-  TerminusOptionsFactory,
-  DNSHealthIndicator,
-  TerminusModuleOptions
-} from '@nestjs/terminus';
+@@filename(health.controller)
+import { HealthCheckService } from '@nestjs/terminus';
 import { Injectable } from '@nestjs/common';
 import { DogHealthIndicator } from './dog.health';
 
 @Injectable()
-export class TerminusOptionsService implements TerminusOptionsFactory {
+export class HealthController {
   constructor(
+    private readonly health: HealthCheckService,
     private readonly dogHealthIndicator: DogHealthIndicator
   ) {}
 
-  createTerminusOptions(): TerminusModuleOptions {
-    const healthEndpoint: TerminusEndpoint = {
-      url: '/health',
-      healthIndicators: [
-        async () => this.dogHealthIndicator.isHealthy('dog'),
-      ],
-    };
-    return {
-      endpoints: [healthEndpoint],
-    };
+  @Get()
+  @HealthCheck()
```

Comment:
Hmm.. Could we assume that every healthcheck = GET route handler?

Reply:
In normal cases, yes. Are you thinking about collapsing `HealthCheck()` with `Get()`? I thought of that too, but maybe there are some edge cases or you want to use the health result in another way, so I thought I'd keep it as it is. Also, I could imagine that it looks like less magic, since it is very familiar to a Nest user to use the `Get()` decorator.
It's in the end just on more line :)

Reply:
Got it. What I'm trying to say is that *maybe* it would be worth considering to implicitly assume that each health check is in fact a GET endpoint to [1] avoid decorators hell [2] suggest "best-practices". To cover edge cases, `HealthCheck` could accept an options object, e.g. `@ HealthCheck({ method: RequestMethod.POST })`. Just my two cents, ignore if you don't like it!